service/controller/app/v1/resource/chart/desired.go

```diff
@@ -17,6 +17,15 @@ import (
 )
 
 func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
+	deleted, err := key.IsDeleted(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	if deleted {
+		return nil, nil
```

Comment:
That does not make sense at all and when you test this you will notice that you just return the desired state in `newDeleteChange`, which now is nil. So nothing will be deleted in fact. `GetDesiredState` should not be changed at all IMO.